### PR TITLE
Edited based on comments and fixed some bugs

### PR DESCRIPTION
**Pull request by afiller1**
As I outlined in #838 I have:
* added the export to excel feature
* figured out how to connect the official database to GRNsight (we still need to toggle stuff on the EC2 so it can run eternally on beta)
* populated this database
* made it so the checkbox to enable node coloring is initially unchecked (this was decently difficult because there are a few different ways in which node coloring is tracked and I had to figure out how to make it so the unchecked box meant the node coloring wouldn't happen, while also figuring out how to change it from the default of coloring automatically)
* made it so the color legend bar as well as its headers, options, and buttons, don't show up until node coloring is happening (goes along with a comment made in #837 ) -- I thought this was better than just having the bar constantly there because there seem to be some more complex methods attached to producing the color bar that I didn't want to disrupt
* adjusted "no expression data" warning to mention the expression database coloring option
* got rid of the weird extension that @dondi pointed out last week that was at the end of my API call
* node coloring using the db now works on the demo sets interestingly enough
* took care of the other notes @kdahlquist made on #837 
* made it so coloring a workbook that already has expression data using data from the database works
* made it so top and bottom dataset differentiation can happen smoothly using expression db data

I have a couple bugs that I'm still working on. Namely:
* when loaded after a wkbk with expression data, node coloring on a sheet without expression data using expression db has challenges
* need to make it so that, when a wkbk without expression data is colored using the db, it doesn't have to reload the expression data each time something is changed (switching to grid layout, for instance)

I thought I'd handled this first bug, but it came back when I refactored things a little to try and fix the second bug. Beyond this, I'm sure there will be other specific bugs that will make themselves clearer as we keep testing things out. Overall, the node coloring works very well and is decently quick. We will also just need to connect the GRNsight RDS to beta and everything should be good to go!

### CODE REVIEW

File: web-client/public/js/update-app.js [hunk 1]
```diff
@@ -447,14 +483,7 @@ const resetDatasetDropdownMenus = (network) => {
...
     // Add expression database options
-    grnState.nodeColoring.nodeColoringOptions.push({value: "Barreto_2018_wt"});
-    grnState.nodeColoring.nodeColoringOptions.push({value: "Dahlquist_2018_dcin5"});
-    grnState.nodeColoring.nodeColoringOptions.push({value: "Dahlquist_2018_dgln3"});
-    grnState.nodeColoring.nodeColoringOptions.push({value: "Dahlquist_2018_dhap4"});
-    grnState.nodeColoring.nodeColoringOptions.push({value: "Dahlquist_2018_dzap1"});
-    grnState.nodeColoring.nodeColoringOptions.push({value: "Dahlquist_2018_wt"});
-    grnState.nodeColoring.nodeColoringOptions.push({value: "Kitagawa_2002_wt"});
-    grnState.nodeColoring.nodeColoringOptions.push({value: "Thorsen_2007_wt"});
+    expressionDBDatasets.forEach(option => grnState.nodeColoring.nodeColoringOptions.push({value: [option]}));
```
Comment (dondi) [inline web-client/public/js/update-app.js:486]: Nice change! Makes things more data-driven. That way, additions to `expressionDBDatasets` in the future will automatically expand the list of values.

File: web-client/public/js/update-app.js [hunk 2]
```diff
@@ -613,61 +645,117 @@ export const updateApp = grnState => {
...
+            } else {
+                // Code makes it here, but node coloring fails to occur. Why???
+                grnState.nodeColoring.nodeColoringEnabled = true;
+                updaters.renderNodeColoring();
+                grnState.nodeColoring.nodeColoringEnabled = true;
+                $(LOG_FOLD_CHANGE_MAX_VALUE_CLASS).removeClass("hidden");
+                $(LOG_FOLD_CHANGE_MAX_VALUE_SIDEBAR_BUTTON).removeClass("hidden");
+                $(LOG_FOLD_CHANGE_MAX_VALUE_HEADER).removeClass("hidden");
+            }
+        }
```
Comment (dondi) [inline web-client/public/js/update-app.js:757]: Hmmm this sequence is getting a little long, and I see a few repeated patterns. Might be good to consolidate some sequences into functions then call them when needed.

File: web-client/public/js/update-app.js [hunk 3]
```diff
@@ -215,22 +220,49 @@ const buildTimepointsString = function (selection) {
...
+                contentType: false,
+                type: "GET",
+                crossDomain: true
+            }) :
+            $.getJSON(fullUrl)
+            ).done((expressionData) => {
+                resolve(expressionData);
+            }).error(console.log("Error in accessing expression database. Result may just be loading."));
+    });
+
```
[the remark this answers is not in the document]
Comment (afiller1) [reply web-client/public/js/update-app.js:260]: I had async/await syntax initially, but (in reference to my last pull request) it caused such a fuss with our linter that we decided to do then/catch syntax. @dondi

[the remark this answers is not in the document]
Comment (dondi) [reply web-client/public/js/update-app.js:260]: Ohhhhh I had forgotten about that. Yes never mind then; we'll have to upgrade our linter rules sometime.

File: web-client/public/js/update-app.js [hunk 4]
```diff
@@ -215,22 +220,49 @@ const buildTimepointsString = function (selection) {
...
+    return genes.substring(0, genes.length - 1);
+};
+
 const buildURL = function (selection) {
     return selection.timepoints ?
-    "expressiondb?dataset=" + selection.dataset + "&timepoints=" + buildTimepointsString(selection)
-    : "expressiondb?dataset=" + selection.dataset;
+    "expressiondb?dataset=" + selection.dataset + "&genes=" +
+    buildGeneQuery() + "&timepoints=" + buildTimepointsString(selection)
+    : "expressiondb?dataset=" + selection.dataset + "&genes=" + buildGeneQuery();
```
[the remark this answers is not in the document]
Comment (afiller1) [reply web-client/public/js/update-app.js:233]: This is quite lovely!

**Pull request merged**